arduino-ide-extension/src/browser/dialogs/settings/settings.tsx

```diff
@@ -278,9 +278,9 @@ export class SettingsService {
     // after saving all the settings, if we need to change the language we need to perform a reload
     // Only reload if the language differs from the current locale. `nls.locale === undefined` signals english as well
     if (
-      currentLanguage !== nls.locale &&
-      !(currentLanguage === 'en' && nls.locale === undefined)
+      currentLanguage !== (await this.localizationProvider.getCurrentLanguage())
     ) {
+      await this.localizationProvider.setCurrentLanguage(currentLanguage);
       if (currentLanguage === 'en') {
         window.localStorage.removeItem(nls.localeId);
       } else {
```

Comment:
@kittaakos do you think it's necessary to save the locale id to the local storage like this in order for the localization to work?

I thought that setting the current language to the localization provider was enough, so I've tried removing this code, but then the localization didn't work correctly, therefore it appears it is not enough.